docs/esp32/tutorial/udp_multicast.rst

```diff
@@ -0,0 +1,141 @@
+.. _esp32_udp_multicast:
+
+UDP MULTICAST
+=============
+
+Use UDP MULTICAST connections to communicate between different networks.
+
+* udp_multicast_client.py::
+
+    import network, socket, errno, struct, time
+
+    # connect to AP
+    SSID = 'SSID'
+    PASSWORD = 'PASSWORD'
+
+    wlan = network.WLAN(network.STA_IF)
+    wlan.ifconfig(('172.16.55.55', '255.255.255.0', '172.16.55.1', '172.16.55.1'))  # class B network
+    wlan.active(True)
+    wlan.connect(SSID, PASSWORD)
+    while not wlan.isconnected():
+        pass
+
+    print("udp_multicast_client.py")
+    # send and receive
+
+    def inet_aton(str_addr):
+        return bytes(map(int, str_addr.split(".")))
+
+    #TIMEOUT = None  # block
+    #TIMEOUT = 5  # s
+    TIMEOUT = 0  # non blocking
+
+    IP = '224.0.0.111'  # MULTICAST
+    PORT = 5555
+    sockaddr = (IP, PORT)
+
+    wlan = network.WLAN(network.STA_IF)
+    ip = wlan.ifconfig()[0]
+
+    skt = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)  # UDP
+    skt.settimeout(None)
+    skt.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    skt.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, struct.pack(">4s4s", inet_aton(IP), inet_aton(ip)))  # join to the multicast address
+    skt.bind(sockaddr)  # not skt.connect(sockaddr)
+    skt.settimeout(TIMEOUT)
+
+    t = 0
+    while True:
+        try:
+            if time.time() - t >= 3:
+                t = time.time()
+                str_to_send = f'ip:{ip}\t mac:{wlan.config("mac")}'
```

Comment:
You can format the MAC address for readability: `':'.join(map('{:02x}'.format, mac))`

Reply:
Done